src/multilspy/language_server.py

```diff
@@ -1539,43 +1559,72 @@ def load_cache(self):
                 # cache often becomes corrupt, so just skip loading it
                 self.logger.log(
                         f"Failed to load document symbols cache from {self._cache_path}: {e}. Possible cause: the cache file is corrupted. " 
-                        "Check for any errors related to saving the cache in the logs.", 
+                        "Check for any errors related to saving the cache in the logs.",
                         logging.ERROR
                     )
 
 
+    async def request_workspace_symbol(self, query: str) -> Union[List[multilspy_types.UnifiedSymbolInformation], None]:
+        """
+        Raise a [workspace/symbol](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_symbol) request to the Language Server
+        to find symbols across the whole workspace. Wait for the response and return the result.
+
+        :param query: The query string to filter symbols by
+
+        :return Union[List[multilspy_types.UnifiedSymbolInformation], None]: A list of matching symbols
+        """
+        response = await self.server.send.workspace_symbol({"query": query})
+        if response is None:
+            return None
+
+        assert isinstance(response, list)
+
+        ret: List[multilspy_types.UnifiedSymbolInformation] = []
+        for item in response:
+            assert isinstance(item, dict)
+
+            assert LSPConstants.NAME in item
+            assert LSPConstants.KIND in item
+            assert LSPConstants.LOCATION in item
+
+            ret.append(multilspy_types.UnifiedSymbolInformation(**item))
+
+        return ret
+
 @ensure_all_methods_implemented(LanguageServer)
 class SyncLanguageServer:
     """
     The SyncLanguageServer class provides a language agnostic interface to the Language Server Protocol.
     It is used to communicate with Language Servers of different programming languages.
     """
 
-    def __init__(self, language_server: LanguageServer) -> None:
+    def __init__(self, language_server: LanguageServer, timeout: Optional[int] = None):
         self.language_server = language_server
         self.loop = None
         self.loop_thread = None
-        
+        self.timeout = timeout
+
         self._server_context = None
 
     @classmethod
     def create(
         cls, config: MultilspyConfig, logger: MultilspyLogger, repository_root_path: str, add_gitignore_content_to_config=True,
+        timeout: Optional[int] = None
```

Comment:
Docstring missing

Reply:
The multilspy devs did not add it. Why do we care?

Reply:
I actually hate it when programmers don't add a unit to this type of parameter. I suppose it's in seconds, but you never know without reading the code.

Reply:
Added docstrings.